engines/private/private.cpp

```diff
@@ -0,0 +1,1086 @@
+#include "audio/audiostream.h"
+#include "common/archive.h"
+#include "common/config-manager.h"
+#include "common/debug.h"
+#include "common/debug-channels.h"
+#include "common/error.h"
+#include "common/events.h"
+#include "common/file.h"
+#include "common/savefile.h"
+#include "common/system.h"
+#include "common/str.h"
+#include "common/timer.h"
+#include "engines/util.h"
+#include "image/bmp.h"
+
+#include "private/cursors.h"
+#include "private/private.h"
+#include "private/tokens.h"
+#include "private/grammar.h"
+
+namespace Private {
+
+PrivateEngine *g_private = NULL;
+
+extern int parse(char *);
+
+PrivateEngine::PrivateEngine(OSystem *syst, const ADGameDescription *gd)
+    : Engine(syst), _gameDescription(gd) {
+    _rnd = new Common::RandomSource("private");
+
+    // Debug channels
+    DebugMan.addDebugChannel(kPrivateDebugFunction, "functions", "Function execution debug channel");
+    DebugMan.addDebugChannel(kPrivateDebugCode, "code", "Code execution debug channel");
+    DebugMan.addDebugChannel(kPrivateDebugScript, "script", "Script execution debug channel");
+
+    // Global object for external reference
+    g_private = this;
+
+    // Setting execution
+    _nextSetting = NULL;
+    _currentSetting = NULL;
+    _pausedSetting = NULL;
+    _modified = false;
+    _mode = -1;
+    _toTake = false;
+
+    // Movies
+    _nextMovie = NULL;
+    _nextVS = NULL;
+    _repeatedMovieExit = "";
+
+    // Save and load
+    _saveGameMask = NULL;
+    _loadGameMask = NULL;
+
+    // Interface
+    _framePath = "inface/general/inface2.bmp";
+
+    // Police
+    _policeBustEnabled = false;
+    _policeBustSetting = NULL;
+    _numberClicks = 0;
+    _sirenSound = "po/audio/posfx002.wav";
+
+    // General sounds
+    _globalAudioPath = "global/audio/";
+    _noStopSounds = false;
+
+    // Radios and phone
+    _policeRadioArea = NULL;
+    _AMRadioArea = NULL;
+    _phoneArea = NULL;
+    // TODO: use this as a default sound for radio
+    _infaceRadioPath = "inface/radio/";
+    _phonePrefix = "inface/telephon/";
+    _phoneCallSound = "phone.wav";
+
+    // Dossiers
+    _dossierPage = 0;
+    _dossierSuspect = 0;
+    _dossierNextSuspectMask = NULL;
+    _dossierPrevSuspectMask = NULL;
+
+    // Diary
+    _diaryLocPrefix = "inface/diary/loclist/";
+}
+
+PrivateEngine::~PrivateEngine() {
+    // Dispose your resources here
+    delete _frame;
+    delete _rnd;
+
+    // Remove all of our debug levels
+    DebugMan.clearAllDebugChannels();
+}
+
+void PrivateEngine::initializePath(const Common::FSNode &gamePath) {
+    SearchMan.addDirectory(gamePath.getPath(), gamePath, 0, 10);
+}
+
+void PrivateEngine::setNextSetting(Common::String *_ns) {
+    delete _nextSetting;
+    _nextSetting = _ns;
+}
+
+void PrivateEngine::setNextMovie(Common::String *_nm) {
+    delete _nextMovie;
+    _nextMovie = _nm;
+}
+
+void PrivateEngine::setOrigin(const int point[2]) {
+    delete _origin;
+    _origin = new Common::Point(point[0], point[1]);;
```

Comment:
This was removed and _origin is not longer a pointer. `_origin` used to be a pointer because it needs to be `const`, but allowing to replace it by certain values. This is not required anymore.